pkg/controllers/state/cluster.go

```diff
@@ -293,6 +296,35 @@ func (c *Cluster) Reset() {
 	c.nameToProviderID = map[string]string{}
 	c.bindings = map[types.NamespacedName]string{}
 	c.antiAffinityPods = sync.Map{}
+	c.daemonSetPods = sync.Map{}
+}
+
+func (c *Cluster) GetDaemonSetPods(daemonset *appsv1.DaemonSet) *v1.Pod {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if pod, ok := c.daemonSetPods.Load(client.ObjectKeyFromObject(daemonset)); ok {
+		return pod.(*v1.Pod)
```

Comment:
This is a reference to the internal data object, which is dangerous, should use DeepCopy()